### ISSUE TITLE
Merge mining support

### ISSUE THREAD

**Issue by TreeNode** [fancoder/cryptonote-universal-pool#5]
Hi, are there any plans to add support for Merge Mining? Such as what MCN or FCN supports? Thank you!

**Comment by mapleshadow** [comment]
Where did you know I was doing this work?

2014-08-13 23:14 GMT+08:00 TreeNode notifications@github.com:

> Hi, are there any plans to add support for Merge Mining? Such as what MCN
> or FCN supports? Thank you!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fancoder/cryptonote-universal-pool/issues/5.

## 

—枫影—

**Comment by TreeNode** [comment]
So you are working on a cryptonote merge mining pool? I followed the pool's software and see who was forking it. There is more work in progress with merge mining more than 2 coins, for example mining MCN, FCN, and XMR at the same time.

**Comment by mapleshadow** [comment]
I had this idea and plans
But has not yet started writing the relevant code

2014-08-14 3:46 GMT+08:00 TreeNode notifications@github.com:

> So you are working on a cryptonote merge mining pool? I followed the
> pool's software and see who was forking it. There is more work in progress
> with merge mining more than 2 coins, for example mining MCN, FCN, and XMR
> at the same time.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fancoder/cryptonote-universal-pool/issues/5#issuecomment-52100398
> .

## 

—枫影—

**Comment by TreeNode** [comment]
It would be a fantastic addition to add in the merge mining. The only pool I know that allows this right now is minergate.  If you added support, merge mining could be pushed out to many pools. Thank you for your contributions!

**Comment by mapleshadow** [comment]
OK~~~Whether it has the support of all CryptoNight algorithm coins?
Whether coins length is 99 digits or 97 digits

2014-08-14 22:46 GMT+08:00 TreeNode notifications@github.com:

> It would be a fantastic addition to add in the merge mining. The only pool
> I know that allows this right now is minergate. If you added support, merge
> mining could be pushed out to many pools. Thank you for your contributions!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fancoder/cryptonote-universal-pool/issues/5#issuecomment-52192682
> .

## 

—枫影—

**Comment by mapleshadow** [comment]
Also, I have a very good code, I know it says it is not very ethical, Oh ~
~ ~
GPU mining tools of the fees charged, can be replaced mineralization in the
mine pool pool creator own wallet account.
This code, I need to share it?

2014-08-17 11:21 GMT+08:00 枫影 mapleshadow@gmail.com:

> OK~~~Whether it has the support of all CryptoNight algorithm coins?
> Whether coins length is 99 digits or 97 digits
> 
> 2014-08-14 22:46 GMT+08:00 TreeNode notifications@github.com:
> 
> It would be a fantastic addition to add in the merge mining. The only pool
> 
> > I know that allows this right now is minergate. If you added support, merge
> > mining could be pushed out to many pools. Thank you for your contributions!
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/fancoder/cryptonote-universal-pool/issues/5#issuecomment-52192682
> > .
> 
> ## 
> 
> —枫影—

## 

—枫影—

**Comment by fancoder** [comment]
Hi guys!
I'm working on merge mining right now. Coming out this or next week.

**Comment by TreeNode** [comment]
Sounds great! Thank you!

**Comment by fancoder** [comment]
Guys, what did you mean merge mining? CryptoNote currencies do not support merge mining for now. I'm going to implement some kind of dinamical pool mining multiple coins and automatically switching to most profitable(hash/USD). It will take some time, I think 2-4 weeks.

**Comment by TreeNode** [comment]
You merge both coins at once, not split hash rate. This is possible. Minergate.com has figured it out. 

As CyptoMidnight explains:

Merged mining works like this: you have two totally separate block chains, they are not related in any way nor does either contain any data from the other. When you mine you generate hashes that may be the solution to the current block, this is very very improbable per hash, its like a lottery where everyone generates tickets until someone finds the winning one. 

Normally you make tickets and check them against the BCN/QCN/XMR or other CN coin block chain to see if they are the solution. With merged mining you create a ticket and check it against both the any CN-based coin block chain and the MonetaVerde block chain,  CN based coin and MCN know nothing about each other, they are two totally different lotteries with different winning numbers, you just sent a copy of your ticket to both. Since you are sending the same ticket to two lotteries you increase your chances of winning one or the other. 

Coins remain totally separate, you simply run both the MonetaVerde and oter CN coin clients on the same machine and submit hashes to both networks. If your hash is the solution to the Monetaverde block you get monetaverdes if you hash is the solution to the other coin block you get both coins. It's exactly like if you where mining on just one network, except you submit the same work twice.

**Comment by netmonk** [comment]
What i dont understand is that the data sent to miner is a blob related to a block header. 

How come the same blockheader +nonce can produce a valid block on multiple blockchain ? 

The current blockheader wich is currently mined on monero has nothing to do with the current blockheader in monetaverde or duckcoin !

Please enlight me

**Comment by mapleshadow** [comment]
Good idea！

2014-08-27 3:48 GMT+08:00 Netmonk notifications@github.com:

> What i dont understand is that the data sent to miner is a blob related to
> a block header.
> 
> How come the same blockheader +nonce can produce a valid block on multiple
> blockchain ?
> 
> The current blockheader wich is currently mined on monero has nothing to
> do with the current blockheader in monetaverde or duckcoin !
> 
> Please enlight me
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fancoder/cryptonote-universal-pool/issues/5#issuecomment-53477878
> .

## 

—枫影—

**Comment by fancoder** [comment]
> What i dont understand is that the data sent to miner is a blob related to a block header.

@netmonk I also do not understand how it works. I'll try to check how minergate works.

**Comment by TreeNode** [comment]
Hi, has there been any progress? Did you talk to a rep from minergate?

**Comment by fancoder** [comment]
Yes, I did. They said that it's proprietary technology, and they don't want to share it to open source.

**Comment by TreeNode** [comment]
Is the code in minerc (for MCN) not helpful in figuring this out? https://github.com/cryptomidnight/monetaverde/blob/master/src/minerc/minerc.cpp#L131

Possible some more code:
https://bitcointalk.org/index.php?topic=675107.msg7655813#msg7655813

Thank you again for your efforts!

**Comment by TreeNode** [comment]
Well, somebody figured it out other than minergate. It's obvious since only half of the hashrate is now at minergate while the other half is somewhere else. Which makes me wonder if someone figured it out and are abusing it. For example, they are using merge mining on their pool to squeeze out extra coins (MCN and FCN) from unbeknownst miners that don't realize they are merge-mining.

**Comment by eightsixeight** [comment]
well thats an understatement "For example, they are using merge mining on their pool to squeeze out extra coins (MCN and FCN) from unbeknownst miners that don't realize they are merge-mining."

of course 90% of XMR miners don't even know they mining, it's all abused systems ran by bots, so minergate is just cashing in on mergemining as to alleviate the cost of the bot installs

**Comment by M7rc7ry** [comment]
Hi. Did you guys ever make any headway with merged mining of cryptonote coins?

**Comment by nibbons** [comment]
I am looking for Cryptonote-util for XDN merged mining fork.

**Comment by crappyrules** [comment]
It appears that mergedmining is supported by this pool code, but how does the configuration change to support both coin daemons?
